microsetta_public_api/resources_alt.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict
+
+from microsetta_public_api.backend import constructors
+from microsetta_public_api.config import (
+    SERVER_CONFIG,
+)
+
+
+class Component:
+
+    def __init__(self, name):
+        self.name = name
+        self.data = None
+        self.children: Dict[str, 'Component'] = dict()
+        self.parent = None
+
+    def add_child(self, child: 'Component'):
+        self.children.update({child.name: child})
+        return child
+
+    def set(self, other: 'Component'):
+        self.__dict__ = other.__dict__
```

Comment:
Is this a standard pattern in python?  It looks like a copy constructor, but instead of being a copy constructor, is a copy... setter?  Under what circumstances do you want to set a node in a tree to be a copy of another node (presumably in the same tree???)

Reply:
Eh maybe this pattern is a little dumb since python isn't type safe :D 
The intended use is basically that I want to pass data around with `resources_alt`, and `set` provides a way to update the data, something like
```
from resources_alt import resources_alt
comp = Component('some data here')
# change the contents of resources_alt, and these changes can be accessed by other modules
resources_alt.set(comp)
```

It would probably be better for me to do something like:
```
def set(self, other: 'Component'):
    self.name = other.name
    self.data = other.data
    self.children = other.children
    self.parent = other.parent
```

I am not sure what to do with parent since this could get tricky.... may delete since I don't really use the parent anywhere, anyways.

Reply:
With the current code you could do something like:
```
from resources_alt import resources_alt
class SomeObject:
    pass
obj = SomeObject()
resources_alt.set(obj)
```
and break everything :D I should change this.

Reply:
Fixed in https://github.com/biocore/microsetta-public-api/pull/57/commits/bf278b80b5b898af365f5590f1bf2d8ca7254b32